internal/service/hosts/hosts.go

```diff
@@ -29,12 +26,7 @@ func New(c *biz.HostUsecase) *HostsService {
 
 func (c *HostsService) CreateRhelHost(ctx context.Context, r *pb.CreateRhelHostRequest) (*pb.CreateRhelHostResponse, error) {
 	if err := r.ValidateAll(); err != nil {
-		return nil, err
-	}
-
-	//TODO: refactor / abstract resource type strings
-	if !strings.EqualFold(r.Host.Metadata.ResourceType, biz.ResourceType) {
```

Comment:
Removing this validation, `ResourceType` is an output only data so it doesn't make sense to test as input.